go/connector/run.go

```diff
@@ -225,38 +225,40 @@ func runCommand(
 		},
 	}
 
+	var group = sync.WaitGroup{}
 	var conn net.Conn
 	// If port is specified, use TCP socket
 	if port != "" {
+		// We're going to need to await 3 separate async tasks:
+		// The attempt to dial, and the two copy operations in
+		// and out of the established connection.
+		group.Add(3)
```

Comment:
If you use `errgroup` instead you can skip the manual adding and subtracting, although I'm not sure how that would interact with the `fe.onError()` abstraction so feel free to ignore if that seems like too much work.

Reply:
The reason for not using `errgroup` is that we don't actually want to await the _first_ error, but rather the completion of _all_ the async tasks, regardless of the result.

Reply:
I'm not sure if I understand the problem, `errgroup.Wait()` always waits until all tasks have terminated, it just returns the first non-nil error (if any) that was returned by one of the tasks.

Reply:
Apparently I'd misunderstood the docs on that, but it seems you're right. I'm inclined to still keep the existing `sync.WaitGroup` though, because the docs for `errgroup` don't specify the behavior in the case that more goroutines are added after the call to `Wait`. This condition seems unlikely, but I couldn't actually prove to myself that it couldn't happen in practice. The problem case would be that the connector starts up and then crashes before we start the goroutines to copy. I could still definitely refactor this and get `errgroup` working, but it doesn't seem worth the effort at this point.